### PR TITLE
Pw optimize csv to object service

### PR DESCRIPTION
**Pull request by Ayuan100**
Could use the CSVToObjectService for every table now.

### CODE REVIEW

File: src/main/java/edu/ucsb/cs56/ucsb_open_lab_scheduler/services/CSVToObjectService.java [hunk 1]
```diff
@@ -8,17 +8,16 @@
...
+public interface CSVToObjectService<T> {
+    List<T> parse(Reader csv, Class<T> type);
 
     Logger getLogger();
 
-    default <T extends Object> List<T> parse(MultipartFile file) {
+    default List<T> parse(MultipartFile file, Class<T> type) {
         try {
-            return parse(new InputStreamReader(file.getInputStream()));
+            return parse(new InputStreamReader(file.getInputStream()), type);
```
Comment (scottpchow23) [inline src/main/java/edu/ucsb/cs56/ucsb_open_lab_scheduler/services/CSVToObjectService.java]: `new InputStreamReader(...)` opens a stream but never closes it.

Comment (Ayuan100) [reply src/main/java/edu/ucsb/cs56/ucsb_open_lab_scheduler/services/CSVToObjectService.java]: Fixed

File: src/main/java/edu/ucsb/cs56/ucsb_open_lab_scheduler/services/CSVToObjectService.java [hunk 2]
```diff
@@ -8,17 +8,19 @@
...
 
     Logger getLogger();
 
-    default <T extends Object> List<T> parse(MultipartFile file) {
+    default List<T> parse(MultipartFile file, Class<T> type) {
         try {
-            return parse(new InputStreamReader(file.getInputStream()));
+            InputStreamReader reader = new InputStreamReader(file.getInputStream());
+            List<T> result = parse(reader, type);
+            reader.close();
```
Comment (btk5h) [inline src/main/java/edu/ucsb/cs56/ucsb_open_lab_scheduler/services/CSVToObjectService.java]: This does not close the reader in all cases. The close statement must be put in a finally block or preferably, a [try-with-resources block](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html) should be used when creating the reader.

Comment (Ayuan100) [reply src/main/java/edu/ucsb/cs56/ucsb_open_lab_scheduler/services/CSVToObjectService.java]: Thanks! Updated.

**Pull request merged**